collectors/feed/collectors.py

```diff
@@ -0,0 +1,281 @@
+        ]
+        return new_feed_entries
+
+    def get_feed_data(self):
+        """Get data for new entries for a particular feed and return FeedEntry instances."""
+        feed_entries = self.get_new_feed_entries()
+
+        feed_models = []
+        for entry in feed_entries:
+            # Special rules below to ignore certain posts.
+
+            # Google Chrome Releases: receives posts for all Chrome products and branches,
+            # but we're only interested in the Stable Channel updates. Skip the rest.
+            if "GoogleChromeReleases" in self.url and CHROMIUM_REGEX.match(entry.title):
+                continue
+
+            # Ignore RHSAs in the Packet Storm advisory feed
+            if (
+                "packetstormsecurity" in self.url
+                and "Red Hat Security Advisory" in entry.title
+            ):
+                continue
+
+            log.info("Processing feed entry: " + entry.link)
+
+            # Certain feeds do not include their full text in the feed entries. For these,
+            # call the get_full_text() function to extract it from the feed directly.
+            special_feeds = [
+                "Packet Storm Advisory Files",
+            ]
+            text = None
+            skip_processing = False
+            try:
+                if self.feed_name in special_feeds:
+                    text = self.get_full_feed_text(entry.link)
+                else:
+                    if "summary" not in entry:
+                        log.info(
+                            "Failed (no summary) to get feed text entry for: %s",
+                            entry.link,
+                        )
+                        # TODO: Consider retrieving text using modified get_full_feed_text for node.js
+                        skip_processing = True
+                    else:
+                        text = html_to_text(entry.summary)
+            except Exception:
+                log.exception(
+                    "Failed (processing error) to get feed entry text for: %s",
+                    entry.link,
+                )
+                skip_processing = True
+
+            if not text:
+                # TODO: Consider retrieving text using modified get_full_feed_text for node.js
+                log.info("No text for feed entry: %s", entry.link)
+                skip_processing = True
+
+            if skip_processing:
+                # Prevent reprocessing next time
+                log.info("Skipping feed entry: %s", entry.link)
+                feed_models.append(
+                    SkippedFeedEntry(
+                        url=entry.link,
+                        skipped=True,
+                    )
+                )
+                continue
+
+            cves = find_cves(entry.title + text)
+            cwes = find_cwes(entry.title + text)
+            cvss = find_cvss(text)
+
+            blocklist, allowlist = check_keywords(entry.title + " " + text)
```

Comment:
I think we do not need `blocklist`. Instead of that, we can just easily ignore the non-matched items.

Reply:
I'll perform measurements, but even before that, I don't think that would work. For example, the allowlist contains the word "kernel". Reading through the blocklist, there's a number of things that interact with a kernel in an irrelevant way, have their own irrelevant version of the Linux kernel, or even have their own non-Linux kernel.

Reply:
Without the blocklist, today's run is not affected. With creating Snippets only for things that explicitly trigger the allowlist (like you do in https://github.com/RedHatProductSecurity/osidb/pull/374 ) it skips everything.

Since the task is about porting the collector and not about reconsidering what exact data it fetches, and the original collector apparently took a lot of time to fine-tune w.r.t. the filtering, I think any functional change whatsoever in the filtering algorithm is out of scope. Not pursuing this further.

Reply:
> With creating Snippets only for things that explicitly trigger the allowlist (like you do in #374 ) it skips everything.

How? You want to create a snippet if a description contains any of the allowed keywords, and these are specified in the allowlist. Am I missing something?

Reply:
If having the blocklist and not having it gives the same results then throw it away. If not and it was in the original Assembler logic then do not change the logic and only migrate it. Any changes of the logic should be subjects of the future enhancements not to unnecessarily complicate the migration.

Reply:
> You want to create a snippet if a description contains any of the allowed keywords, and these are specified in the allowlist.

This is not how the feed collector in Assembler works. It generates snippets/tickets for everything by default, and skips feed entries with at least one blocklisted word iff that entry contains zero allowlisted words.

> Am I missing something?

Either of us is missing something, but IDK who. I _think_ the goal of the task is to port the collector and have the flaw drafts/snippets created directly instead of through Assembler so that the effective end result is similar from IR point of view, and I _think_ the goal is not changing how the collector works by e.g. eliminating the majority of what it generates.

Reply:
Saw Ondrej's message only after posting my reply to Jitka. Closing the thread then.